provisioning/windows-provision.ps1

```diff
@@ -338,11 +338,17 @@ Write-Output "= Windows Powershell & Powershell Core sanity checks:"
 Invoke-Command {& "C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -command "(Get-Host).Version"}
 Invoke-Command {& "C:\Program Files\PowerShell\7\pwsh.exe" -command "(Get-Host).Version"}
 
-# Install Vagrant (requires chocolatey)
+# Install Tools with Chocolatey Packages
 Write-Output "= Installing Vagrant..."
-
 Invoke-Command {& "choco.exe" install vagrant --yes --no-progress --limit-output --fail-on-error-output --version $env:VAGRANT_VERSION;}
 
+Write-Output "= Installing Ruby..."
+# Append a ".1" as all ruby packages in chocolatey have this suffix. Not sure why (maybe a package build id)
+Invoke-Command {& "choco.exe" install ruby --yes --no-progress --limit-output --fail-on-error-output --version "${env:RUBY_VERSION}.1";}
+Write-Host '- Sanity check for ruby tooling'
+& C:\tools\ruby26\bin\ruby -v
+& C:\tools\ruby26\bin\bundle -v
+

```

Comment:
Yes: 
- As chocolatey is needed, we have to ensure the correct order. Adding an element after the block l243 would be fragile: anyone reordering or renaming this code would make this fail.
- The struct l243 is built to "download -> (eventually) unarchive -> install -> (eventually) run post commands -> (eventually) add to path. I felt like it was not fit to run a chocolatey (or set of) chocolatey commands only.

=> Potential improvement for later: maybe the same kind of struct bu for chocolatey packages, or adapt the struct for chocolatey packages. I'm not willing to do this because it would be too much work for doing the same thing as these 2 lines.

Reply:
> As chocolatey is needed, we have to ensure the correct order. Adding an element after the block l243 would be fragile: anyone reordering or renaming this code would make this fail.

> The struct l243 is built to "download -> (eventually) unarchive -> install -> (eventually) run post commands -> (eventually) add to path. I felt like it was not fit to run a chocolatey (or set of) chocolatey commands only.

Right. Would have liked this feedback while adding choco ^^ I would have gone the same way here as the other since this issue already exists as it is.
WDYT of installing choco on its own first, then all these tools with a specific choco data structure?

Reply:
To be quite honest, I did not caught it during the review of chocolatey and only realize it in this PR

Reply:
https://github.com/jenkins-infra/packer-images/issues/270 ❤️